hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/EstimatePrecompileFeature.java

```diff
@@ -1614,6 +1502,250 @@ public void exchangeRateTinyBarsToTinyCentsEstimateGas() {
         validateGasEstimation(data, EXCHANGE_RATE_TINYBARS_TO_TINYCENTS, estimatePrecompileContractSolidityAddress);
     }
 
+    private void executeContractTransaction(
+            DeployedContract deployedContract,
+            int gas,
+            ContractMethods contractMethods,
+            ContractFunctionParameters parameters) {
+
+        ExecuteContractResult executeContractResult = contractClient.executeContract(
+                deployedContract.contractId(), gas, contractMethods.getSelector(), parameters, null);
+
+        networkTransactionResponse = executeContractResult.networkTransactionResponse();
+        verifyMirrorTransactionsResponse(mirrorClient, 200);
+    }
+
+    public int validateAndReturnGas(String data, ContractMethods contractMethods, String contractAddress) {
+        var response = estimateContract(data, contractAddress);
+        var estimateGasValue = response.getResultAsNumber().intValue();
+        assertTrue(isWithinDeviation(contractMethods.getActualGas(), estimateGasValue, lowerDeviation, upperDeviation));
+        return estimateGasValue;
+    }
+
+    @Then("I call estimateGas with balanceOf function for {string} and verify the estimated gas against HAPI")
+    public void executeBalanceOfFunctionWithLimitedGas(String tokenName) {
```

Comment:
We should avoid manual conversion to enum.

```suggestion
    @Then("I call estimateGas with balanceOf function for {token} and verify the estimated gas against HAPI")
    public void executeBalanceOfFunctionWithLimitedGas(TokenNameEnum tokenName) {
```

Reply:
Removed the manual conversion.